LivingCity/traffic/b18CommandLineVersion.cpp

```diff
@@ -169,7 +180,126 @@ void B18CommandLineVersion::runB18Simulation() {
+
+          std::getline(ss, value, ','); // Read route_id
+          try {
+              busRouteIds.push_back(std::stoi(value));
+          } catch (const std::invalid_argument& e) {
+              std::cerr << "Invalid route_id: " << value << " in line: " << line << std::endl;
+              continue;
+          }
+
+          std::getline(ss, value, ','); // Skip route_short_name
+
+          std::getline(ss, value, ','); // Read departure_time
+          try {
+              busDepartureTimes.push_back(std::stoi(value));
+          } catch (const std::invalid_argument& e) {
+              std::cerr << "Invalid departure_time: " << value << " in line: " << line << std::endl;
+              continue;
+          }
+
+          std::getline(ss, value, '"'); // 跳过第一个引号前的内容
+          std::getline(ss, value, '"'); // 读取引号之间的内容
+          std::vector<int> stops;
+
+          // 处理引号和方括号
+          value.erase(std::remove(value.begin(), value.end(), '['), value.end());
+          value.erase(std::remove(value.begin(), value.end(), ']'), value.end());
+
+          std::stringstream stopStream(value);
+          std::string stop;
+          while (std::getline(stopStream, stop, ',')) {
+              // 去除空格并解析
+              stop.erase(0, stop.find_first_not_of(' '));
+              stop.erase(stop.find_last_not_of(' ') + 1);
+              try {
+                  stops.push_back(std::stoll(stop));
+              } catch (const std::invalid_argument& e) {
+                  std::cerr << "Invalid stop_id: " << stop << " in line: " << line << std::endl;
+              }
+          }
+          busRoutes.push_back(stops);
+
+          // 读取 routing 字段（也包含在引号内）
+          std::getline(ss, value);
+          std::vector<int> routing;
+
+          // 处理引号和方括号
+          value.erase(std::remove(value.begin(), value.end(), '"'), value.end());
+          value.erase(std::remove(value.begin(), value.end(), '['), value.end());
+          value.erase(std::remove(value.begin(), value.end(), ']'), value.end());
+
+          std::stringstream routingStream(value);
+          std::string route;
+          while (std::getline(routingStream, route, ',')) {
+              // 去除空格并解析
+              route.erase(0, route.find_first_not_of(' '));
+              route.erase(route.find_last_not_of(' ') + 1);
+              if (route.empty() || !std::all_of(route.begin(), route.end(), ::isdigit)) {
+                  //std::cerr << "Invalid routing_id: " << route << " in line: " << line << std::endl;
+                  continue;
+              }
+              try {
+                  routing.push_back(std::stoll(route));
+              } catch (const std::invalid_argument& e) {
+                  //std::cerr << "Invalid routing_id: " << route << " in line: " << line << std::endl;
+              }
+          }
+          busRoutings.push_back(routing);
+      }
+
+
+  }
+
+for (size_t i = 0; i < busRoutes.size(); ++i) {
```

Comment:
@SLZ0106  This should be reading trip ID, and use the new bus_shcedule file Junzhe